test/getcookie.js

```diff
@@ -0,0 +1,93 @@
+
+process.env.NO_DEPRECATION = 'express-session';
+
+var after = require('after')
+var assert = require('assert')
+var express = require('express')
+  , request = require('supertest')
+  , cookieParser = require('cookie-parser')
+  , session = require('../')
+  , Cookie = require('../session/cookie')
+var fs = require('fs')
+var http = require('http')
+var https = require('https')
+var util = require('util')
+
+var min = 60 * 1000;
+
+describe('session()', function(){
+  
+     
+  it('should allow a custom getcookie and setcookie option', function(done){
+    var server = createServer(createServer({ 
+      name:'foo', 
+      secret: 'bar',
+      getcookie(req, name, secrets) {
+        var cookies = cookie.parse(headers.authorization || '');
+        console.log('getting cookie');
+        return signature.unsign(cookies[name] || '', secrets[0]);
+      },
+      setcookie(res, name, val, secret, options) {
+        var signed = signature.sign(val, secret);
+        var data = cookie.serialize(name, signed, options);
+        res.setHeader('authorization', data);
+        console.log('Setting header');
+      }
+    }), function (req, res) {
+      req.session.user = 'John';
+      res.end()
+    })
+    request(server)
+    .get('/admin')
+    .expect(function (res) {
+      console.log(res.headers)
```

Comment:
I'm not familiar with this testing,  this console.log returns   `{ 'set-cookie': [ 'connect.sid=s%3Ax2fXsI8gGwByJoPat0MFajRByCYqGLTl.%2BswySoAbDN7t5XAf25VvXOrCRLMi53vlreAQ5atKEzc; Path=/; Expires=Sun, 29 Jan 2017 15:03:19 GMT; HttpOnly' ],
  date: 'Sun, 29 Jan 2017 15:02:20 GMT',
  connection: 'close',
  'content-length': '0' }` I don't understand yet why it doesn't do in the custom setcookie function